mitmproxy/proxy/protocol/base.py

```diff
@@ -107,9 +107,14 @@ def __check_self_connect(self):
         """
         address = self.server_conn.address
         if address:
+            forbidden_hosts = ["localhost", "127.0.0.1", "::1"]
+
+            if self.config.options.listen_host:
+                forbidden_hosts.append(self.config.options.listen_host)
```

Comment:
Could we not just always include `self.config.options.listen_host`? If that is empty, the `address[0]` check should also never be true.

Reply:
That's preference I guess and it was my very first thought as well. But I don't like having an empty string in the `forbidden_hosts` list. It just doesn't feel right. But sure the check would still work because `address[0]` won't be an empty string. I just like to write very explicit code that doesn't require a second thought when looking at it. E.g. when someone later looks at it they might think "wait a second, can't listen_host be empty?" and they need to put work into understanding that for the end result it doesn't matter. Basically I try to prevent humans from playing computer in their heads.

Reply:
Fair enough, I'm convinced. Thanks! 🍰